qcodes/dataset/data_set.py

```diff
@@ -251,10 +254,30 @@ def paramspecs(self) -> Dict[str, ParamSpec]:
         return dict(zip(param_names, params))
 
     @property
-    def exp_id(self):
+    def exp_id(self) -> int:
         return select_one_where(self.conn, "runs",
                                 "exp_id", "run_id", self.run_id)
 
+    @property
+    def exp_name(self) -> str:
+        return get_experiment_name_from_experiment_id(self.conn, self.exp_id)
+
+    @property
+    def sample_name(self) -> str:
+        return get_sample_name_from_experiment_id(self.conn, self.exp_id)
+
+    @property
+    def run_timestamp_raw(self) -> float:
+        """Returns run timestamp as number of seconds since the Epoch"""
```

Comment:
It's probably good to mention (here or elsewhere) when the timestamp is actually made. I guess this one is at the start of the measurement?

Reply:
yes. when the new `run` (which means "a new dataset") is created in the database, a timestamp of that moment is added to the `run` (see the "results" table in the database). That's why the docstring says `run timestamp`. I will methion that it is the "moment when the measurement run started".

Reply:
Great. How about a `completed_timestamp` too?

Reply:
ok, i can add it. anything else? :)

Reply:
added :)